util/src/main/java/com/farao_community/farao/util/SensitivitySecurityAnalysisService.java

```diff
@@ -0,0 +1,139 @@
+
+                    SensitivityComputationResults sensiResults = runSensitivityComputation(network, crac, twoWindingsTransformers);
+                    contingencySensitivityComputationResultsMap.put(contingency, sensiResults);
+
+                    variantsPool.releaseUsedVariant(workingVariant);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            })).get();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } catch (ExecutionException e) {
+            throw new FaraoException(e);
+        }
+        network.getVariantManager().setWorkingVariant(initialVariantId);
+
+        //return SensitivitySecurityAnalysisResult
+        return new SensitivitySecurityAnalysisResult(precontingencyResult, contingencySensitivityComputationResultsMap);
+    }
+
+    public static void applyCracContingency(Network network, ComputationManager computationManager, Contingency contingency) {
+        contingency.getNetworkElements().forEach(contingencyElement -> applyCracContingencyElement(network, computationManager, contingencyElement));
+    }
+
+    public static void applyCracContingencyElement(Network network, ComputationManager computationManager, NetworkElement contingencyElement) {
+        Identifiable element = network.getIdentifiable(contingencyElement.getId());
+        if (element instanceof Branch) {
+            BranchContingency contingency = new BranchContingency(contingencyElement.getId());
+            contingency.toTask().modify(network, computationManager);
+        } else {
+            throw new FaraoException("Unable to apply contingency element " + contingencyElement.getId());
+        }
+    }
+
+    public static List<TwoWindingsTransformer> getPstInRangeActions(Network network, List<RangeAction> rangeActions) {
+        List<TwoWindingsTransformer> psts = new ArrayList<>();
+        for (RangeAction rangeAction : rangeActions) {
+            List<NetworkElement> networkElements = rangeAction.getNetworkElements();
+//            LOGGER.info("In RangeAction: " + rangeAction.getId());
+            for (NetworkElement networkElement : networkElements) {
+                if (isPst(network, networkElement)) {
+//                    LOGGER.info("adding " + networkElement.getId());
+                    psts.add(network.getTwoWindingsTransformer(networkElement.getId()));
+                } else {
+                    LOGGER.warn("not support type of range action");
+                }
+            }
+        }
+        return psts;
+    }
+
+    public static SensitivityComputationResults runSensitivityComputation(
+            Network network,
+            Crac crac,
+            List<TwoWindingsTransformer> psts) {
+        SensitivityFactorsProvider factorsProvider = net -> {
+            List<SensitivityFactor> factors = new ArrayList<>();
+            crac.getCnecs().forEach(cnec -> {
```

Comment:
After crac refactoring #99 you will be able to select cnec by state, meaning preventive or after contingency. So you'll have to refactor this a bit but it can be made in a future PR.

Reply:
No action for now.